observability/micrometer/src/main/java/org/wildfly/extension/micrometer/registry/ApplicationRegistry.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.extension.micrometer.registry;
+
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.ToDoubleFunction;
+import java.util.stream.StreamSupport;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.search.RequiredSearch;
+import io.micrometer.core.instrument.search.Search;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+
+public class ApplicationRegistry extends SimpleMeterRegistry {
```

Comment:
@jasondlee MeterRegistry has other methods that are not accounted for here.

1) clear().
2) config(). And then the API exposed by MeterRegistry.Config.

I think clear() could be implemented, e.g forEachMeter(ApplicationRegistry::remove)

MeterRegistry.Config is more complicated. Some things, like Config.meterFilter and the other methods that call it, I don't think can be supported. Others I don't know but I think there are other problems, like reachable things that expose a 'shutdown' method. It might be better to just log a WARN if that is called.